src/controllers/connectionManager.ts

```diff
@@ -672,6 +673,27 @@ export default class ConnectionManager {
     // create a new connection with the connectionCreds provided
     public async connect(fileUri: string, connectionCreds: IConnectionInfo, promise?: Deferred<boolean>): Promise<boolean> {
         const self = this;
+        if (connectionCreds.authenticationType === Constants.azureMfa) {
+            if (!connectionCreds.azureAccountToken) {
```

Comment:
What if the token exists but is expired?

Reply:
Yes, once the existing token gets expired, all later requests using that token will fail. What I did with Query Editor in ADS was to store the expiry timestamp along with the token itself. We could do something similar here and refresh token if the token doesn't exist  or token is expired.

Reply:
we definitely need to port of the renew token fix from ADS since that bug significantly impacts the AAD querying experience.  Though if that bug is unrelated to the bug being resolved with this PR it's probably fine to take that as a follow-up PR.